### PR TITLE
Update generate.py to generate opengraph tags

### PR DESCRIPTION
**Pull request by Mariatta**

### CODE REVIEW

File: web/README.md [hunk 1]
```diff
@@ -6,3 +6,16 @@ Go to the `generate_site/` directory,
...
 files for the web.
+
+## Adding Opengraph Meta tag
+
+When adding a new page to the site, please be sure to update the OpenGraph
+meta tag. These tags are useful when sharing the links on social media,
+it generates a nice preview of the site.
+
+To add these info, see the examples of the og_ meta tags in generate.py ``templates``
+variable where we added ``og_title``, etc. to the ``about`` and ``index`` pages.
```
Comment (trallard) [inline web/README.md:17]:
```suggestion
To add this info, refer to the `og_meta` tags defined for the current templates (about, index) in `generate.py`.
```
 
I found the previous wording a bit confusing, so this is more of an optional suggestion than a hard-needed change.

File: web/generate_site/generate.py [hunk 2]
```diff
@@ -19,15 +19,36 @@
...
+    _t = environment.from_string(
+        open(f"templates/{key}-base.html").read()
+    )
+    template_params = conf
+    template_params.update(value)
     # Write final HTML
-    with open(Path("..") / f"{_template}.html", "w") as f:
-        f.write(_t.render(conf))
+    with open(Path("..") / f"{key}.html", "w") as f:
+        print(f"writing {f=}")
```
Comment (trallard) [inline web/generate_site/generate.py:52]: Thank you 🙏🏽 - knowing when things are happening is useful

File: web/generate_site/generate.py [hunk 3]
```diff
@@ -19,15 +19,36 @@
...
 
-templates = ["about", "index"]
-environment = Environment(loader=FileSystemLoader("templates/"))
+templates = {"about":
+    {
+        "og_title": "PyLadiesCon 2023 - About Us",
+        "og_description": "About PyLadiesCon.",
+        "og_type": "article",
+        "og_url": "https://conference.pyladies.com/about.html",
+        "og_image_url": "https://conference.pyladies.com/img/icon.png",
```
Comment (trallard) [inline web/generate_site/generate.py:31]:
```suggestion
        "og_image_url": "https://conference.pyladies.com/img/icon.png",
        "og_image_alt": "PyLadiesCong logo',
```
Let's add alt text to our images to make these accessible to screen readers and other assistive tech

File: web/generate_site/generate.py [hunk 4]
```diff
@@ -19,15 +19,36 @@
...
+    },
+    "index": {
+        "og_title": "PyLadiesCon 2023",
+        "og_description": "PyLadiesCon 2023. Global PyLadies Community Conference.",
+        "og_type": "website",
+        "og_url": "https://conference.pyladies.com/index.html",
+        "og_image_url": "https://conference.pyladies.com/img/icon.png",
```
Comment (trallard) [inline web/generate_site/generate.py:38]:
```suggestion
        "og_image_url": "https://conference.pyladies.com/img/icon.png",
        "og_image_alt": "PyLadiesCong logo',
```

File: web/generate_site/templates/head.html [hunk 5]
```diff
@@ -2,7 +2,12 @@
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
-    <title>PyLadiesCon 2023</title>
+    <meta property="og:title" content="{{ og_title }}">
+    <meta property="og:description" content="{{ og_description }}">
+    <meta property="og:type" content="{{ og_type }}">
+    <meta property="og:url" content="{{ og_url }}">
+    <meta property="og:image" content="{{ og_image_url }}">
```
Comment (trallard) [inline web/generate_site/templates/head.html:9]:
```suggestion
    <meta property="og:image" content="{{ og_image_url }}">
    <meta property="og:image:alt" content="{{ og_image_alt }}">
```

Ref: https://ogp.me/#structured

**Review by trallard — CHANGES_REQUESTED** [review]

**Comment by trallard** [comment]
Thanks, @Mariatta - I added some comments and suggestions considering that all our images have non-descriptive names such as "icon" it is important to add some alt-text to make the content accessible.

**Comment by Mariatta** [comment]
Thanks! Please re-review. I committed the changes and fixed some typos.

**Review by trallard — APPROVED** [review]
Looks great, thanks ✨

**Pull request merged**